extension/scripts/Selector/SelectorDocument.js

```diff
@@ -0,0 +1,110 @@
+var SelectorDocument = {
+    canReturnMultipleRecords: function () {
+        return true;
+    },
+
+    canHaveChildSelectors: function () {
+        return false;
+    },
+
+    canHaveLocalChildSelectors: function () {
+        return false;
+    },
+
+    canCreateNewJobs: function () {
+        return false;
+    },
+
+    willReturnElements: function () {
+        return false;
+    },
+
+    _getData: function (parentElement) {
+
+        var elements = this.getDataElements(parentElement);
+
+        var dfd = $.Deferred();
+
+        // return empty record if not multiple type and no elements found
+        if (this.multiple === false && elements.length === 0) {
+            var data = {};
+            data[this.id] = null;
+            dfd.resolve([data]);
+            return dfd;
+        }
+
+        // extract links one by one
+        var deferredDataExtractionCalls = [];
+        $(elements).each(function (k, element) {
+
+            deferredDataExtractionCalls.push(function (element) {
+
+                var href = element.href;
+                if (this.stringReplacement && this.stringReplacement.replaceString) {
+                    var replace;
+                    var replacement = this.stringReplacement.replacementString || "";
+                    try {
+                        var regex = new RegExp(this.stringReplacement.replaceString, 'gm');
+                        replace = regex.test(href) ? regex : this.stringReplacement.replaceString;
+                    } catch (e) {
+                        replace = this.stringReplacement.replaceString;
+                    }
+
+                    href = href.replace(replace, replacement);
+                }
```

Comment:
I'd suggest to move string replacement in SelectorLink, SelectorDocument and SelectorImage to Selector class.